datacite-draft-doi-handler/src/main/java/no/unit/nva/doi/DraftDoiHandler.java

```diff
@@ -0,0 +1,134 @@
+package no.unit.nva.doi;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import java.io.IOException;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Optional;
+import javax.xml.bind.JAXBException;
+import no.unit.nva.doi.datacite.clients.DataCiteClient;
+import no.unit.nva.doi.datacite.clients.exception.ClientException;
+import no.unit.nva.doi.datacite.clients.models.Doi;
+import no.unit.nva.doi.datacite.config.DataCiteConfigurationFactory;
+import no.unit.nva.doi.datacite.config.PasswordAuthenticationFactory;
+import no.unit.nva.doi.datacite.mdsclient.DataCiteMdsConnectionFactory;
+import no.unit.nva.events.handlers.DestinationsEventBridgeEventHandler;
+import no.unit.nva.events.models.AwsEventBridgeDetail;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import no.unit.nva.publication.doi.dto.Publication;
+import no.unit.nva.publication.doi.dto.PublicationHolder;
+import no.unit.nva.publication.doi.update.dto.DoiUpdateDto;
+import no.unit.nva.transformer.Transformer;
+import no.unit.nva.transformer.dto.DataCiteMetadataDto;
+import nva.commons.utils.IoUtils;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DraftDoiHandler extends DestinationsEventBridgeEventHandler<PublicationHolder, DoiUpdateDto> {
+
+    // exception messages
+    public static final String PUBLICATION_IS_MISSING_ERROR = "Publication is missing";
+    public static final String CUSTOMER_ID_IS_MISSING_ERROR = "CustomerId is missing";
+    public static final String TRANSFORMING_PUBLICATION_ERROR = "Error transforming Publication to DataCite XML";
+
+    // log messages
+    public static final String RECEIVED_REQUEST_TO_CREATE_DRAFT_NEW_DOI_LOG =
+        "Received request to create draft new DOI for {}";
+    public static final String DRAFTED_NEW_DOI_LOG = "Drafted new DOI: {}";
+    public static final String ERROR_DRAFTING_DOI_LOG = "Error drafting DOI ";
+
+    private DoiClient doiClient;
+
+    private static final Logger logger = LoggerFactory.getLogger(DraftDoiHandler.class);
+
+    /**
+     * Default constructor for DraftDoiHandler.
+     *
+     * @throws IOException  IOException
+     */
+    @JacocoGenerated
+    public  DraftDoiHandler() throws IOException {
+        this(defaultDoiClient());
+    }
+
+    @JacocoGenerated
+    private static DataCiteConfigurationFactory defaultConfigFactory() {
+        String dataCiteConfigJson = AppEnv.getDataCiteConfig();
+        return new DataCiteConfigurationFactory(IoUtils.stringToStream(dataCiteConfigJson));
```

Comment:
Here you transform a string to stream to string. Is there any reason?

Reply:
We can make that constructor in `DataCiteConfigurationFactory(String)` public, I made it protected to ensure I tested the InputStream method and got coverage.

Reply:
Updated to use new constructor